Library/Homebrew/extend/os/linux/extend/ENV/shared.rb

```diff
@@ -7,8 +7,10 @@ def effective_arch
       @bottle_arch.to_sym
     elsif @build_bottle
       Hardware.oldest_cpu
-    else
+    elsif Hardware::CPU.intel? || Hardware::CPU.arm?
       :native
+    else
+      :none
     end
```

Comment:
This should work fine for PPC too.  Changing this also breaks `HOMEBREW_ARCH` (though maybe that doesn't matter much).

An alternative one-liner could be to modify `arch_flag` to return nothing for `native` if `CPU.type == :dunno`?